inst/tutorials/tidy-02-causality2/tidy-02-causality2.Rmd

```diff
@@ -0,0 +1,286 @@
+
+
+### Using simple conditional statements
+
+For this exercise, we'll use the resume data once again.
+
+What if we wanted to create a new vector that depends on whether a statement is true or false? For example, suppose you wanted to create an indicator variable for whether or not a specific resume had the name "Carrie." We can  create a new variable using the `if_else(X, Y, Z)` command. This command takes a logical vector as `X` and returns a new vector of the same length as `X` that has the value `Y` if that value in `X` is TRUE and `Z` if that value in `X` is FALSE.
+
+
+#### Exercise
+
+- Use the `if_else` function to create a new variable called `carrie` that is 1 if the resume name (`firstname`) is `"Carrie"` and 0 otherwise.
+- Print the first six lines of `resume` using the `head` function to see the new variable.
+
+```{r ifelse, exercise = TRUE}
+## create a new variable called carrie
+## print the first 6 lines of the updated resume
+resume %>%
+  mutate(carrie = )
+
+```
+
+```{r ifelse-hint}
+## create a new variable called carrie
+## print the first 6 lines of the updated resume
+resume %>%
+  mutate(carrie = if_else(???, ???, ???)) %>%
+  ???()
+```
+
+
+```{r ifelse-solution}
+## create a new variable called carrie
+## print the first 6 lines of the updated resume
+resume %>%
+  mutate(carrie = if_else(firstname == "Carrie", 1, 0)) %>%
+  head()
+```
+
+```{r ifelse-check}
+grade_result_strict(
+  pass_if(~ identical(resume$carrie, if_else(resume$firstname == "Carrie", 1, 0)))
+)
+```
+
+## Factors variables in R
+
+### Factor variables
+
+
+You have seen that creating subsets can be helpful for calculating different quantities or statistics for specific subgroups in the data. When there is more than 1 or 2 subgroups of interest, however, this can be a cumbersome process. For that reason, it's helpful to know about factor variables. Basically, a factor variable is a categorical variable that takes a finite number of distinct values.
+
+Any variable can be turned into a factor by calling the `as.factor()` function like so:
+
+    mydata$myvar <- as.factor(mydata$myvar)
+
+This will take the variable `myvar` and create a factor variable with levels that are observed in that variable. Most often, you will convert a character variable to a factor.
+
+#### Exercise
+
+- Finish the code below that creates the `type` character variable. Fill in the last values of `race` and `sex` and add the label `WhiteMale` to this last type.
+- Convert the `type` variable to a factor variable using `mutate()` and `as.factor()`.
+
+```{r factor-setup}
+resume <-
+  resume %>%
+  mutate(carrie = if_else(firstname == "Carrie", 1, 0))
+```
+
+```{r factor, exercise = TRUE}
+## fill in the last line of code to create a character vector for the type of
+## application that was sent
+## turn the character vector into a factor
+resume <- resume %>%
```

Comment:
Might it be cleaner to use `case_when` here?

Reply:
That's right. I kept this exercise, while adding one more exercise with `case_when`, given that the textbook highlights both ways.